### DISCUSSION TITLE
Difference between Swarms and Ray?

### DISCUSSION THREAD

**Discussion by theo-dim** [kyegomez/swarms#433] [General]
I am very early in the process of identifying tools to parallelize certain inference tasks at work and we have been using Ray internally to scale up an algo we made in-house, on-prem (algo trading, if anyone is active in the area). Ray is very tailored towards ML workloads, which to my understanding is only one of the types of tasks Swarms can handle. Is it correct that Swarms handles a broader superset of tasks in parallel?

Thanks team I appreciate it,
Theo

**Comment by kyegomez** [comment]
@theo-dim 

Swarms is for multi-agent collaboration, across a vast array of different communication architectures. 

Ray is for training llms on multiple gpus.

very different.
